### PR TITLE
Update More_Prefix_Sums.mdx

### PR DESCRIPTION
**Pull request by Aryansh-S**
Added difference arrays

### CODE REVIEW

File: content/3_Silver/More_Prefix_Sums.mdx [hunk 1]
```diff
@@ -3,16 +3,52 @@ id: more-prefix-sums
...
+
+$P^{-1}(A)$ is sometimes referred to as the **difference array** of $A.$ (And in C++, just as the standard library has the builtin prefix sum function `partial_sum`, it has `adjacent_difference` for difference arrays.) 
+
+For example, the difference array of $\{2, 5, 9, 6, 8\}$ is $\{2, 3, 4, -3, 2\},$ as the prefix sum of $\{2, 3, 4, -3, 2\}$ is $\{2, 5, 9, 6, 8\}.$
+
+Though the utility of difference arrays may not be immediately evident, consider the following somewhat standard problem (with variations that have come up on numerous past silver contests).
+
+### Problem - Range Update Queries
+
+You are given an array $A$ of size $n.$ There are $q$ queries of the form $(i,j,v)$ indicating that you must increment each element in the range $A[i] \dots A[j]$ by $v,$ where $0 \leq i \leq j < n.$ After all $q$ queries, you are asked to compute $A.$
```
Comment (SansPapyrus683) [inline content/3_Silver/More_Prefix_Sums.mdx:28]: think it's better to use N and Q (uppercase) for consistency

File: content/3_Silver/More_Prefix_Sums.mdx [hunk 2]
```diff
@@ -3,16 +3,52 @@ id: more-prefix-sums
...
+
+Constraints: $1 \leq n,q \leq 10^5$
```
Comment (SansPapyrus683) [inline content/3_Silver/More_Prefix_Sums.mdx:30]: put this in parenthesis maybe?
"we have an array of size n (1 <= n <= 1e5)"

File: content/3_Silver/More_Prefix_Sums.mdx [hunk 3]
```diff
@@ -3,16 +3,52 @@ id: more-prefix-sums
...
+Consider the difference array $P^{-1}(A).$ Each query of incrementing from $A[i]$ to $A[j]$ by $v$ is equivalent to the following two increments in the difference array:
+* $P^{-1}(A)[i]$ by $v$
+* $P^{-1}(A)[j+1]$ by $-v$
+
+Then, whenever we choose to recompute $A$ as $P(P^{-1}(A)),$ the increment of $v$ will simply propagate across the range $A[i] \dots A[j]$ and stop when it reaches $A[j+1].$
+
+Overall, our difference array solution is $\mathcal{O}(n+q),$ fast enough to pass:
+* Precompute $P^{-1}(A)$ in $\mathcal{O}(n)$ 
+* Perform each of $q$ queries in $\mathcal{O}(1),$ for $\mathcal{O}(q)$ overall 
+* Prefix sum to tabulate all the queries at the end in $\mathcal{O}(n)$
```
Comment (SansPapyrus683) [inline content/3_Silver/More_Prefix_Sums.mdx:49]: 1. maybe have some solution code? (i can add it on the cf gym maybe if need be)
2. but anyways, what's this for?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Comment by SansPapyrus683** [comment]
are we assuming people have done all the problems in the previous modules? i still think it's fine tbh

**Comment by bqi343** [comment]
> are we assuming people have done all the problems in the previous modules?

Yes, it's starred.

**Comment by HarryWangATX** [comment]
Yes, I don't think this will be necessary either.

**Pull request closed without merging**